docs/ref/logging.txt

```diff
@@ -0,0 +1,380 @@
+
+When :setting:`DEBUG` is ``True``:
+
+* The ``django`` logger sends messages in the ``django`` hierarchy (except
+  ``django.server``) at the ``INFO`` level or higher to the console.
+
+When :setting:`DEBUG` is ``False``:
+
+* The ``django`` logger sends messages in the ``django`` hierarchy (except
+  ``django.server``)  with ``ERROR`` or ``CRITICAL`` level to
+  :class:`AdminEmailHandler`.
+
+Independently of the value of :setting:`DEBUG`:
+
+* The :ref:`django-server-logger` logger sends messages at the ``INFO`` level
+  or higher to the console.
+
+All loggers except :ref:`django-server-logger` propagate logging to their
+parents, up to the root ``django`` logger. The ``console`` and ``mail_admins``
+handlers are attached to the root logger to provide the behavior described
+above.
+
+Default logging definition
+--------------------------
+
+The default configuration is available as ``django.utils.log.DEFAULT_LOGGING``
+and defined in :source:`django/utils/log.py`::
+
+	{
+	    'version': 1,
+	    'disable_existing_loggers': False,
+	    'filters': {
+	        'require_debug_false': {
+	            '()': 'django.utils.log.RequireDebugFalse',
+	        },
+	        'require_debug_true': {
+	            '()': 'django.utils.log.RequireDebugTrue',
+	        },
+	    },
+	    'formatters': {
+	        'django.server': {
+	            '()': 'django.utils.log.ServerFormatter',
+	            'format': '[{server_time}] {message}',
+	            'style': '{',
+	        }
+	    },
+	    'handlers': {
+	        'console': {
+	            'level': 'INFO',
+	            'filters': ['require_debug_true'],
+	            'class': 'logging.StreamHandler',
+	        },
+	        'django.server': {
+	            'level': 'INFO',
+	            'class': 'logging.StreamHandler',
+	            'formatter': 'django.server',
+	        },
+	        'mail_admins': {
+	            'level': 'ERROR',
+	            'filters': ['require_debug_false'],
+	            'class': 'django.utils.log.AdminEmailHandler'
+	        }
+	    },
+	    'loggers': {
+	        'django': {
+	            'handlers': ['console', 'mail_admins'],
+	            'level': 'INFO',
+	        },
+	        'django.server': {
+	            'handlers': ['django.server'],
+	            'level': 'INFO',
+	            'propagate': False,
+	        },
+	    }
+	}   
+
+See :ref:`Configuring logging <configuring-logging>` on how to
+complement or replace this default logging configuration.
+
+Django logging extensions
+==================================
+
+Django provides a number of utilities to handle the particular requirements of
+logging in a web server environment.
+
+Loggers
+-------
+
+Django provides several built-in loggers.
+
+.. _django-logger:
+
+``django``
+~~~~~~~~~~
+
+The parent logger for messages in the ``django`` :ref:`named logger
+hierarchy <naming-loggers-hierarchy>`. Django does not post messages using this name; instead it uses one of the loggers below.
```

Comment:
Wrap at 79 chars. I would try to avoid semicolons, can we rewrite the last sentence? e.g.:
 ```suggestion
The parent logger for messages in the ``django`` :ref:`named logger hierarchy
<naming-loggers-hierarchy>`. Django does not post messages using this name.
Instead, it uses one of the loggers below.
```